From e026e4bba247241edd503b1133cd2094d7be7b04 Mon Sep 17 00:00:00 2001
From: David 'Digit' Turner <digit@android.com>
Date: Fri, 26 Mar 2010 13:28:49 -0700
Subject: [PATCH 247/247] cpufreq: fix typo

Signed-off-by: David 'Digit' Turner <digit@android.com>
---
 include/linux/cpufreq.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/linux/cpufreq.h b/include/linux/cpufreq.h
index 83fcfa7..37b9a16 100644
--- a/include/linux/cpufreq.h
+++ b/include/linux/cpufreq.h
@@ -255,7 +255,7 @@ void cpufreq_notify_transition(struct cpufreq_freqs *freqs, unsigned int state);
 #ifdef CONFIG_CPU_FREQ_STAT
 extern void cpufreq_exit_idle(int cpu, unsigned long ticks);
 #else
-#define cpufreq_exit_idle(int cpu, unsigned long ticks) do {} while (0)
+#define cpufreq_exit_idle(cpu, ticks) do {} while (0)
 #endif
 
 static inline void cpufreq_verify_within_limits(struct cpufreq_policy *policy, unsigned int min, unsigned int max) 
-- 
1.7.0.4

